### DISCUSSION TITLE
Support `@googleapis//google/type:date_proto` dependency from `ts_proto_library`

### DISCUSSION THREAD

**Discussion by refactornator** [aspect-build/rules_ts#627] [Q&A]
I have a `proto_library` defined as follows

```
proto_library(
    name = "analytics_proto",
    srcs = ["analytics.proto"],
    deps = [
        "@googleapis//google/type:date_proto",
    ],
)
```

In `analytics.proto` I'm importing Google's Date Proto in the usual way
```
import "google/type/date.proto";
```
And using it in a message.

But, when I then generate my `.ts` stubs using `ts_proto_library`, I get output that tries to import the Date type from a path that does not exist.

```
ts_proto_library(
    name = "analytics_ts_proto",
    node_modules = "//:node_modules",
    proto = ":analytics_proto",
    proto_srcs = [
        "analytics.proto",
    ],
)
```
The broken import is in `analytics_pb.d.ts`
```
import type { Date } from "../../../google/type/date_pb.js";
```

Is this approach supported by `ts_proto_library`?

**Comment by ryanli** [comment]
I'm having the same issue with `@googleapis//google/rpc:status_proto`. I can declare a `ts_proto_library` for the proto itself in my project, e.g.:

```
ts_proto_library(
    name = "status_ts_proto",
    node_modules = "//my/project:node_modules",
    proto = "@googleapis//google/rpc:status_proto",
    gen_connect_es = False,
    copy_files = False,
)
```

However if I try to build it I will get an the following error:

```
Error in declare_file: the output artifact 'external/googleapis~/google/rpc/status_pb.js' is not under package directory 'my/project' for target '//my/project:_status_ts_proto_protoc'
```

**Comment by ramilmsh** [comment]
@refactornator @ryanli 

have you ever find a solution ? i've ran into exactly the same issue with status_pb
